src/record.rs

```diff
@@ -99,13 +121,95 @@ impl<'a> Serializable<'a> for MultiPlexedRecord<'a> {
                 position,
                 payload,
             }),
+            RecordType::AppendRecords => Some(MultiPlexedRecord::AppendRecords {
+                queue,
+                position,
+                records: MultiRecord::new(payload)?,
+            }),
             RecordType::Truncate => Some(MultiPlexedRecord::Truncate { queue, position }),
             RecordType::Touch => Some(MultiPlexedRecord::Touch { queue, position }),
             RecordType::DeleteQueue => Some(MultiPlexedRecord::DeleteQueue { queue, position }),
         }
     }
 }
 
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub struct MultiRecord<'a> {
+    buffer: &'a [u8],
+    position: usize,
+}
+
+impl<'a> MultiRecord<'a> {
+    pub fn new(buffer: &[u8]) -> Option<MultiRecord> {
+        let mut mrecord = MultiRecord {
+            buffer,
+            position: 0,
+        };
+        // verify the content is not corrupted
+        for record in mrecord {
+            record?;
+        }
+
+        mrecord.reset_position();
+
+        Some(mrecord)
+    }
+
+    pub fn new_unchecked(buffer: &[u8]) -> MultiRecord {
+        MultiRecord {
+            buffer,
+            position: 0,
+        }
+    }
+
+    pub fn serialize<'b, T: Iterator<Item = &'b [u8]>>(items: T, mut position: u64) -> Vec<u8> {
+        let mut res = Vec::new();
+        for item in items {
+            res.extend_from_slice(&position.to_le_bytes());
+            position += 1;
+            res.extend_from_slice(&item.len().to_le_bytes());
+            res.extend_from_slice(item);
+        }
+        res
+    }
+
+    pub fn reset_position(&mut self) {
+        self.position = 0;
+    }
+}
+
+impl<'a> Iterator for MultiRecord<'a> {
+    type Item = Option<(u64, &'a [u8])>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.position == self.buffer.len() {
+            // no more record
+            return None;
+        }
+
+        let buffer = &self.buffer[self.position..];
+        if buffer.len() < 16 {
+            // too short: corrupted
+            self.position = buffer.len();
+            return Some(None);
+        }
+
+        let position = u64::from_le_bytes(buffer[0..8].try_into().unwrap());
+        let len = u64::from_le_bytes(buffer[8..16].try_into().unwrap()) as usize;
```

Comment:
I forgot. We don't have a lower limit for a record? u64 seems pretty high.

Reply:
oh yes, we do have a upper limit of u16::MAX bytes per record